openwisp_users/api/permissions.py

```diff
@@ -66,3 +68,32 @@ class IsOrganizationOwner(BaseOrganizationPermission):
 
     def validate_membership(self, user, org):
         return org and (user.is_superuser or user.is_owner(org))
+
+
+class CustomDjangoModelPermissions(DjangoModelPermissions):
+    def __init__(self):
+        self.perms_map = copy.deepcopy(self.perms_map)
+        self.perms_map['GET'] = ['%(app_label)s.view_%(model_name)s']
+
+    def has_permission(self, request, view):
+        # Workaround to ensure DjangoModelPermissions are not applied
+        # to the root view when using DefaultRouter.
+        if getattr(view, '_ignore_model_permissions', False):
+            return True
+
+        if not request.user or (
+            not request.user.is_authenticated and self.authenticated_users_only
+        ):
+            return False
+
+        queryset = self._queryset(view)
+        perms = self.get_required_permissions(request.method, queryset.model)
+        change_perm = self.get_required_permissions('PUT', queryset.model)
+
+        if request.method == 'GET':
+            if request.user.has_perms(perms) or request.user.has_perms(change_perm):
+                return True
+            else:
+                return False
```

Comment:
Did not handled the case of super user becasue superuser by default has all the permissions.

Reply:
I think you can add a check at the top stating:
```
if user.is_superuser:
    return True
```

They are allowed to do anything regardless of any permission!

Reply:
@atb00ker I am trying to implement this in more generic way, cause am thinking of opening a similar PR in DRF repo once this gets merged here. 

By default `superuser` has all permissions i.e., it can do anything it wants. suppose manually someone removes all the permissions for any specific model for `superuser`. Then in that case this would not work, since we will return True if it;s a super user, so that case will fail.

What do you think?

Reply:
To the best of my knowledge, removing permissions from `superuser` is an anti-pattern, but I understand your point.
1. We should look at existing DRF functions and implement it in a similar way.
2. I'll also wait for @nemesisdesign check this! :smile: